src/service/data_products/genome_attributes.py

```diff
@@ -258,19 +295,52 @@ async def _get_filters(
                 raise errors.IllegalParameterError(
                     f"More than one filter specification provided for field {field}")
             filter_query[field] = r.query_params[q]
+    return filter_query
+
+
+async def _get_filters(
+    r: Request,
+    arango_coll: str,
+    coll_id: str,
+    load_ver: str,
+    load_ver_override: bool,
+    view_name: str = None,
+    count: bool = False,
+    sort_on: str = None,
+    sort_desc: bool = False,
+    filter_conjunction: bool = True,
+    match_spec: SubsetSpecification = None,
+    selection_spec: SubsetSpecification = None,
+    keep: dict[str, set[col_models.ColumnType]] = None,
```

Comment:
I thought a column can only have one column type. Why we have a set here?

Reply:
Because the keep parameter specifies the allowable column types, for which there could be > 1

Reply:
ah, I assume the `key` of keep variable refers to a column name? A column's data type should be unique and consistent. Unless it refers to a data model or something else ...

Reply:
> ah, I assume the key of keep variable refers to a column name?

Correct

> A column's data type should be unique and consistent.

Which they are, e.g. https://github.com/kbase/collections/blob/main/src/common/collection_column_specs/genome_attribs-PMI.yml

Reply:
if they are unique and consistent, why each col could have a set of types?

Reply:
It's not the columns having a set of types, it's that the allowed column types for the histogram operation are a set. IOW, if the provided column is in the set {int, float} the operation can proceed, but if not, then an error is thrown. Take a look how the sets are used in the `_get_filters` method.

Reply:
E.g. https://github.com/kbase/collections/pull/533/files#diff-21772263d31206be1473129ce3db2950fcf3be1d52d3f702ea3d75497387f08aR338-R341

Reply:
https://github.com/kbase/collections/pull/533/files#diff-21772263d31206be1473129ce3db2950fcf3be1d52d3f702ea3d75497387f08aR561

Make more sense now. Indeed, It does not conflict with my point: https://github.com/kbase/collections/pull/533#discussion_r1385319584